### PR TITLE
feat/table cleanup

### PR DESCRIPTION
**Pull request by bclark-p44**
DES-61: Updating table design to support pagination and sorting.

Screen Shot 2022-02-28 at 12 17 49 PM

### CODE REVIEW

**Review by eemahoney — CHANGES_REQUESTED** [review]
Thanks for the updates @brandon-convey 

I pulled and served locally came across a few UX issues to address:

* Sort icons should persist on sortable columns. Team has had a lot of discussion so no worries that this got a bit lost. See my comment and discussion thread [here](https://p-44.slack.com/archives/C02HH1T46UF/p1645744619945949?thread_ts=1645739935.795419&cid=C02HH1T46UF).
* Sorting is not keyboard accessible. Focusing and using the Return key to toggle between asc and desc does not work. (Find a good example of that working from the [W3 Table example](https://w3c.github.io/aria-practices/examples/table/sortable-table.html))
* If keyboard accessible, shouldn't the sortable headers be focusable? I _think_ I am seeing the browser/system default focus ring

Screen Shot 2022-02-28 at 12 46 13 PM

Other things to note, perhaps out of the scope of this work:

Shouldn't we have a scroll bar if the table width goes beyond the screen? Currently I'm seeing this when minimizing the screen and I do not have the ability to scroll over to the right end of the table.

Screen Shot 2022-02-28 at 12 57 04 PM

**Comment by eemahoney** [comment]
For Storybook, are we only supposed to see these available props in the Controls panel?
Screen Shot 2022-02-28 at 1 11 12 PM

**Comment by eemahoney** [comment]
Feb-28-2022 13-14-26

Confirming the width change when an icon is added to a header

File: packages/table/src/TableFooter.tsx [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
-})({ display: 'table-footer-group' });
+export type TableFooterProps = ComponentProps<typeof StyledTableFooter>;
 
 export const TableFooter = React.forwardRef<HTMLTableSectionElement, TableFooterProps>(
   (props: TableFooterProps, ref) => (
-    <TableContext.Provider value={{ variant: 'header' }}>
-      <TableFooterRoot ref={ref} {...props} />
-    </TableContext.Provider>
+    <TableCellContext.Provider value={{ variant: 'footer' }}>
+      <StyledTableFooter className="manifestui-table-footer" ref={ref} {...props} />
```
[the remark this answers is not in the document]
Comment (bclark-p44) [reply packages/table/src/TableFooter.tsx:11]: we can override styles via the theme, but this allows for inline overrides via the `sx` prop if a consumer wants.

File: packages/color/package.json [hunk 2]
```diff
@@ -14,5 +14,9 @@
     "directory": "packages/color"
   },
   "author": "project44",
-  "license": "MIT"
+  "license": "MIT",
+  "dependencies": {
```
[the remark this answers is not in the document]
Comment (bclark-p44) [reply packages/color/package.json:18]: nope, these are regular dependencies

File: .storybook/addons/styled/enhancers.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import merge from 'lodash.merge';
+
+const formatStyledProps = ({ argTypes }) => {
```
[the remark this answers is not in the document]
Comment (bclark-p44) [reply .storybook/addons/styled/enhancers.js:3]: We could definitely extend this out in the future. For now, I went with the simplest implementation. I will add some comments

**Comment by bclark-p44** [comment]
@eemahoney 

> For Storybook, are we only supposed to see these available props in the Controls panel?

Those are the only props that extend the native element props. I thought we didn't want to list every HTML prop in this table?

> Confirming the width change when an icon is added to a header

Yes, the sort icon adds width to the header. I can make it absolute and not extend the width if we'd like.

For keyboard accessibility that should be easy enough to add, although we need a design for that. Do we want the whole cell to be focusable? or just the sort button (the text and the icon). If we go the whole-cell route, we are getting into data-grid territory. 

> Shouldn't we have a scroll bar if the table width goes beyond the screen? Currently, I'm seeing this when minimizing the screen and I do not have the ability to scroll over to the right end of the table.

I think we need to decide on paradigms for horizontal scrolling, do we want just the body / header to scroll? Does that include the footer? What about if the footer is pagination?

**Comment by bclark-p44** [comment]
@eemahoney just going to re-hash what we discussed IRL here on the ticket for transparency:

1. For storybook we can't get rid of the `docs` tabs completely because of the need to document sub-component APIs. There is a limitation in SB that will not add sub-component props to the `controls` section in canvas, requiring us to keep this tab around.
2. Design is through the following: Header icon width (whether it should add to the width or not), keyboard accessibility rules, focus states for column sort button, table width, and horizontal scrolling.

For those reasons above, we are going to hold off on merging / making updates to this PR until we've had time to consider some of the lingering questions/discrepancies between design and development.

**Comment by eemahoney** [comment]
Pull and confirmed new changes. Good stuff @brandon-convey, thank you

**Review by eemahoney — DISMISSED** [review]

File: .storybook/addons/styled/enhancers.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (bclark-p44) [reply .storybook/addons/styled/enhancers.js:12]: @eemahoney added some comments here.

**Comment by eemahoney** [comment]
Screen Shot 2022-03-02 at 1 12 33 PM

I might be asking the wrong question but should header cells have a min width? Is this expected? with the header and sorting icon in my screenshot?

**Review by eemahoney — APPROVED** [review]
Thank you for all the hard work on this table @bclark-p44 . Let's :shipit:

**Pull request merged**